Fix `pat-autotoc` fieldset ID and refactor TOC navigation

petschki opened the pull request:
@cihanandac I had to refactor the id generation because I've overseen that the PR #1468 rewrites the `fieldset` ids with auto-generated values (which we only want if we have vertical TOC or colliding fieldset Ids) ... see https://jenkins.plone.org/job/pull-request-6.1-3.10/974/ for robot test details, which depend on various fieldset Ids (eg `#fieldset-settings`)

petschki commented on src/pat/autotoc/autotoc.js:
```diff
@@ -78,16 +78,21 @@ export default Base.extend({
-            $(section).attr("id", id).attr("aria-labelledby", tabId);
+
+            if (!asTabs) {
+                $level.attr("id", id).attr("aria-labelledby", tabId);
```
I set the auto generated ID to the level item, so that its uniquely labelled. In theory, we do not have to trigger the click event anymore, because the URL hash points to the level element. What you think @thet ?

petschki commented on src/pat/autotoc/autotoc.js:
```diff
@@ -78,16 +78,21 @@ export default Base.extend({
+            const $section = $level.closest(self.options.section);
+            let id = $section.prop("id") || $level.prop("id");
 
-            if (!id || $("#" + id).length > 0) {
```
@thet I do not understand the logic behind this. We should not auto-create an ID if we have a unique fieldset#id but for my understanding this if is always true (no ID or an element with ID) ... not really sure how this could work before.

Additionally we can generate this always if we create a vertical TOC and assign it to the level elements to get a unique URL hash.

cihanandac commented:
I was also not aware that there are tests that are checking for the fieldset ids :) 
I have tried this version and it seems like it correctly matches the tab ids with the fieldset counter parts. 

Also, refactored parts looks good and easier to read.

Thanks @petschki

petschki commented:
OK, somehow this autotoc implementation confused me ... sorted some things for clarification:

The old "autotab" implementation iterates over the `levels` selector, generates a `nav > a` tab element with a generated unique ID and adds a click event which is linked to the corresponding section visibility. So the "tab#id" and the "section#id" are completely separated of each other. Furthermore the `section#id` was untouched (this is mandatory for some robottests!) The generated tab markup looked like this:

<img width="760" alt="Bildschirmfoto 2025-06-23 um 15 49 04" src="https://github.com/user-attachments/assets/7753695e-feb2-40c2-b60a-6132a6f3c0ef" />

A bit strange, that the `href` of each tab points to its own id.

Now we have the Bootstrap markup implementation (https://getbootstrap.com/docs/5.3/components/navs-tabs/#javascript-behavior) ... we need to add things like `data-bs-` and `aria-` attributes so that we can use `Bootstrap.Tab` library.

So this markup

```html
<form>
<fieldset id="fieldset-default">
<legend id="fieldsetlegend-default">Default</legend>
</fieldset>
</form>
```

should end in

```html
<form>
<nav>
<ul class="nav nav-tabs" id="myTab" role="tablist">
  <li class="nav-item" role="presentation">
    <a href="#fieldset-default" class="nav-link active" id="autotoc-item-autotoc-0-tab" data-bs-toggle="tab" data-bs-target="#fieldset-default"  role="tab" aria-controls="fieldset-default" aria-selected="true">Default</button>
  </li>
</ul>
</nav>
<div class="tab-content" id="myTabContent">
  <fieldset class="tab-pane fade show active" id="fieldset-default" role="tabpanel" aria-labelledby="autotoc-item-autotoc-0-tab" tabindex="0">...</div>
</div>
<form>
```

Thats straight forward, but currently, there are also robottests which rely on the "old" autogenerated paneID selector (see https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/tests/robot/test_contentbrowser.robot#L126) which needs to be changed accordingly (/cc @1letter)

Theres one more thing: this old issue with the nested tabs and non-unique ID problem https://github.com/plone/mockup/issues/1242 ... maybe someone has an idea how we could handle this?

petschki commented:
I have now reverted the autotoc merge and released mockup 5.4.2 ... this branch is now rebased with the changes from #1468 and I will continue to work on the autotoc fixes.

staticresources with the tinymce fixes from #1476 will get released soon when https://github.com/plone/plone.staticresources/pull/385 is green.

petschki commented on src/pat/autotoc/autotoc.js:
```diff
@@ -98,7 +98,10 @@ export default Base.extend({
+            // NOTE: if you have nested autotocs then you have to add the
+            // parent autotoc tabId to `options.IDPrefix` of the sub autotoc
+            // in order to mark parent and sub tab as active.
+            if (activeId === null && (window.location.hash.indexOf(tabHash) == 0 || $level.hasClass(self.options.classActiveName))) {
```
@cihanandac this solves the nested tab activation problem. however, we need to fix the `IDPrefix` of theming controlpanel before we merge this, otherwise you cannot access the subtabs anymore.
